[NF,NB] array operators in bindings not correct

### Description

When using the element wise multiplication (`.*`) in a binding expression the frontend seems to remove the element-wise part and make it a normal multiplication. This leads to errors when the target code is tried to be compiled.

If the binding is instead an equation it works fine.

### Steps to Reproduce

Use omc with `--newBackend` for following model:
```
model M
  parameter Real[2] x = {1,2};
  parameter Real[2] y = x .* x;
end M;
```

This yields an error:
```
M_06inz.c:26:24: error: incompatible type for argument 1 of ‘simple_array_copy_data’
   26 |   real_array_copy_data(mul_real_scalar_product(tmp0, tmp1), tmp2);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                        |
      |                        modelica_real {aka double}
```
which is a direct result of the multiplication missing the element-wise property.


The following functionally identical model works:
```
model M
  parameter Real[2] x = {1,2};
  parameter Real[2] y(each fixed = false);
initial equation
  y = x .* x;
end M;
```

### Expected Behavior

The frontend should keep the element-wise property of the operator and the model should work the same way the second does.

--- a/OMCompiler/Compiler/NFFrontEnd/NFCeval.mo
+++ b/OMCompiler/Compiler/NFFrontEnd/NFCeval.mo
@@ -982,6 +982,9 @@ algorithm
     case Op.MUL then evalBinaryMul(exp1, exp2);
     case Op.DIV then evalBinaryDiv(exp1, exp2, target);
     case Op.POW then evalBinaryPow(exp1, exp2, target);
+    case Op.ADD_EW then evalBinaryAdd(exp1, exp2);
+    case Op.SUB_EW then evalBinarySub(exp1, exp2);
+    case Op.MUL_EW then evalBinaryMul(exp1, exp2);
     case Op.ADD_SCALAR_ARRAY then evalBinaryScalarArray(exp1, exp2, evalBinaryAdd);
     case Op.ADD_ARRAY_SCALAR then evalBinaryArrayScalar(exp1, exp2, evalBinaryAdd);
     case Op.SUB_SCALAR_ARRAY then evalBinaryScalarArray(exp1, exp2, evalBinarySub);

--- a/OMCompiler/Compiler/NFFrontEnd/NFExpandExp.mo
+++ b/OMCompiler/Compiler/NFFrontEnd/NFExpandExp.mo
@@ -589,7 +589,7 @@ public
       end if;
 
       if expanded then
-        outExp := expandBinaryElementWise2(exp1, op, exp2, SimplifyExp.simplifyBinaryOp);
+        outExp := expandBinaryElementWise2(exp1, Operator.stripEW(op), exp2, SimplifyExp.simplifyBinaryOp);
       else
         outExp := exp;
       end if;

--- a/OMCompiler/Compiler/NFFrontEnd/NFOperator.mo
+++ b/OMCompiler/Compiler/NFFrontEnd/NFOperator.mo
@@ -713,6 +713,19 @@ public
     outOp := OPERATOR(ty, o);
   end makeArrayScalar;
 
+  function makeEW
+    input output Operator op;
+  algorithm
+    () := match op.op
+      case Op.ADD algorithm op.op := Op.ADD_EW; then ();
+      case Op.SUB algorithm op.op := Op.SUB_EW; then ();
+      case Op.MUL algorithm op.op := Op.MUL_EW; then ();
+      case Op.DIV algorithm op.op := Op.DIV_EW; then ();
+      case Op.POW algorithm op.op := Op.POW_EW; then ();
+      else ();
+    end match;
+  end makeEW;
+
   function stripEW
     input output Operator op;
   algorithm

--- a/OMCompiler/Compiler/NFFrontEnd/NFTypeCheck.mo
+++ b/OMCompiler/Compiler/NFFrontEnd/NFTypeCheck.mo
@@ -1294,7 +1294,11 @@ algorithm
       then
         (resultType, op);
 
-    // scalar * scalar and array * array => elemOp.
+    // array * array => Op.{elemOp}_EW
+    case (true, true)
+      then (resultType, Operator.makeEW(Operator.OPERATOR(resultType, elemOp)));
+
+    // scalar * scalar => Op.{elemOp}
     else (resultType, Operator.OPERATOR(resultType, elemOp));
   end match;
 

--- a/testsuite/simulation/modelica/NBackend/differentation/arrayDiff.mos
+++ b/testsuite/simulation/modelica/NBackend/differentation/arrayDiff.mos
@@ -24,12 +24,12 @@ diffSimulationResults("MWE_14574_res.mat", "ReferenceFiles/MWE_14574_res.mat", "
 // true
 // ""
 // ### dumpSimplify | NBSolve.solveBody ###
-// [BEFORE] ({0.0, 0.0} * x .+ y * {0.0, 0.0}) .- (0.0 * $DER.x .+ M * {1.0, 1.0})
+// [BEFORE] ({0.0, 0.0} .* x .+ y .* {0.0, 0.0}) .- (0.0 * $DER.x .+ M * {1.0, 1.0})
 // [AFTER ] -M
 //
 // ### dumpSimplify | NBSolve.solveBody ###
-// [BEFORE] -(y * x .- M * {0.0, 0.0}) / (-M)
-// [AFTER ] (y * x) / M
+// [BEFORE] -(y .* x .- M * {0.0, 0.0}) / (-M)
+// [AFTER ] (y .* x) / M
 //
 // record SimulationResult
 //     resultFile = "MWE_14574_res.mat",

